src/components/ShowList.jsx

```diff
@@ -31,11 +43,38 @@ function ShowList() {
       )}
       {value && value.docs.length > 0 && (
         <div>
-          Collection:
+          <input
+            type="text"
+            placeholder="Filter items..."
+            value={filter}
+            onChange={handleChange}
+          />
+          {filter && (
+            <button
+              onClick={() => setFilter('')}
+              aria-label="clear filter text"
+            >
+              <FontAwesomeIcon icon={['fas', 'times']} />
+            </button>
+          )}
           <ul>
-            {value.docs.map((doc) => (
-              <Item key={doc.id} {...doc.data()} id={doc.id} />
-            ))}
+            {filter
+              ? value.docs
+                  .map((doc) => ({ id: doc.id, ...doc.data() }))
+                  .filter((item) =>
+                    item.name.toLowerCase().includes(filter.toLowerCase()),
+                  )
+                  .map((item) => (
+                    <Item
+                      key={item.id}
+                      id={doc.id}
```

Comment:
What is this doing? I don't see where `doc` is defined.

Reply:
do you mean item.id here?🤔

Reply:
I'm not entirely sure setting the `id` at all is necessary.